torch/csrc/lazy/core/ir_builder.h

```diff
@@ -0,0 +1,109 @@
+#pragma once
+
+#include <c10/core/ScalarType.h>
+#include <c10/util/Optional.h>
+#include <torch/csrc/lazy/backend/backend_interface.h>
+#include <torch/csrc/lazy/core/ir.h>
+#include <vector>
+
+// This file contains autogenerated LazyTensor IrBuilder
+namespace torch {
+namespace lazy {
+
+
+struct IrBuilder {
+  virtual NodePtr MakeDeviceData(const std::shared_ptr<BackendData>& data) const = 0;
+  virtual NodePtr MakeScalar(const at::Scalar& value, const at::ScalarType& type) const = 0;
+  virtual NodePtr MakeExpand(const Value& input0, const std::vector<int64_t>& size, const bool& is_scalar_expand) const = 0;
+  virtual NodePtr MakeView(const Value& input0, const std::vector<int64_t>& output_size) const = 0;
+  virtual NodePtr MakeCast(const Value& input0, const at::ScalarType& dtype, const c10::optional<at::ScalarType>& stype = c10::nullopt) const = 0;
+  virtual NodePtr MakeTensorList(const OpList& inputs) const = 0;
+  virtual NodePtr MakeAsStridedViewUpdate(const Value& input0, const Value& input1, const std::vector<int64_t>& size, const std::vector<int64_t>& stride, const int64_t& storage_offset) const = 0;
+  virtual NodePtr MakeAsStrided(const Value& input0, const std::vector<int64_t>& size, const std::vector<int64_t>& stride, const int64_t& storage_offset) const = 0;
+  virtual NodePtr MakeDiagonalViewUpdate(const Value& input0, const Value& input1, const int64_t& offset, const int64_t& dim1, const int64_t& dim2) const = 0;
+  virtual NodePtr MakeDiagonal(const Value& input0, const int64_t& offset, const int64_t& dim1, const int64_t& dim2) const = 0;
+  virtual NodePtr MakeNarrowViewUpdate(const Value& input0, const Value& input1, const std::vector<int64_t>& base_indices) const = 0;
+  virtual NodePtr MakeNarrow(const Value& input0, const std::vector<int64_t>& base_indices, const std::vector<int64_t>& sizes) const = 0;
+  virtual NodePtr MakePermute(const Value& input0, const std::vector<int64_t>& dims) const = 0;
+  virtual NodePtr MakeResize(const Value& input0, const std::vector<int64_t>& size) const = 0;
+  virtual NodePtr MakeSelectViewUpdate(const Value& input0, const Value& input1, const int64_t& dim, const int64_t& start, const int64_t& end, const int64_t& stride) const = 0;
+  virtual NodePtr MakeSelect(const Value& input0, const int64_t& dim, const int64_t& start, const int64_t& end, const int64_t& stride) const = 0;
+  virtual NodePtr MakeSqueeze(const Value& input0, const int& dim) const = 0;
+  virtual NodePtr MakeUnsqueeze(const Value& input0, const int& dim) const = 0;
+};
+
+static inline NodePtr MakeDeviceData(const std::shared_ptr<BackendData>& data) {
```

Comment:
i wonder if its that useful to keep these helper functions? What about just having a 'GetIrBuilder' helper in here and then replace all the call sites to GetIrBuilder()->Make{OpName}() ?

Reply:
I primarily included them to keep the `core/` core less verbose. e.g. `MakeExpand` is less verbose than `GetIrBuilder()->MakeExpand`